create.ps1

```diff
@@ -1,200 +1,220 @@
-                $LastName += $FamilyName + " - "
-                $LastName += if (-NOT([string]::IsNullOrEmpty($PartnerNamePrefix))) { $PartnerNamePrefix + " " }
-                $LastName += $PartnerName
-            }
-            "PB" {
-                #   $LastName += if (-NOT([string]::IsNullOrEmpty($PartnerNamePrefix))) { $PartnerNamePrefix + " " }
-                $LastName += $PartnerName + " - "
-                $LastName += if (-NOT([string]::IsNullOrEmpty($FamilyNamePrefix))) { $FamilyNamePrefix + " " }
-                $LastName += $FamilyName
-            }
-            Default {
-                #   $LastName += if (-NOT([string]::IsNullOrEmpty($FamilyNamePrefix))) { $FamilyNamePrefix + " " }
-                $LastName += $FamilyName
 
-            }
-        }
-        return $LastName
-            
-    }
-    catch {
-        throw("An error was found in the name convention algorithm: $($_.Exception.Message): $($_.ScriptStackTrace)")
-    } 
-}
+#region functions
 
-switch ($p.details.gender) {
-    "Female" 
-    { $gender = "V" }
-    "Male" 
-    { $gender = "M" }
-    Default 
-    { $gender = "O" }
-}
+#endregion functions
 
-#Change mapping here
-$account = [PSCustomObject]@{
-    email          = $p.Accounts.MicrosoftActiveDirectory.mail;
-    voornaam       = $p.Name.NickName
-    tussenvoegsel  = GenerateMiddleName -person $p
-    achternaam     = GenerateLastName -person $p
-    gebruikersnaam = $p.Accounts.MicrosoftActiveDirectory.SamAccountName
-    externalId     = $p.ExternalId
-    geslacht       = $gender
-}
+try {
+    # Create account object from mapped data and set the correct account reference
+    $account = $actionContext.Data;
+    $person = $personContext.Person;
 
-#correlation
-$correlationField = 'externalId'
-$correlationValue = $p.ExternalID
+    <#
+    # Make sure module is imported
+    $moduleName = "PSSQLite"
 
-Try {     
-     
-    $query = "SELECT $correlationField,achternaam FROM persons WHERE $correlationField = '$correlationValue'"
-    $correlationCheckResult = Invoke-SqliteQuery -Query $query -DataSource $database -Verbose:$verbose
+    # If module is imported say that and do nothing
+    if (Get-Module -Verbose:$false | Where-Object { $_.Name -eq $ModuleName }) {
+        Write-Verbose "Module [$ModuleName] is already imported."
+    }
+    else {
+        # If module is not imported, but available on disk then import
+        if (Get-Module -ListAvailable -Verbose:$false | Where-Object { $_.Name -eq $ModuleName }) {
+            $module = Import-Module $ModuleName -Verbose:$false
+            Write-Verbose "Imported module [$ModuleName]"
+        }
+        else {
+            # If the module is not imported, not available and not in the online gallery then abort
+            throw "Module [$ModuleName] is not available. Please install the module using: Install-Module -Name [$ModuleName] -Force"
```

Comment:
I would suggest to only do 'Import-Module PSSQLite'. First off, the name of the module is fixed and isn't likely to change. Second, 'Import-Module' already throws in case the module cannot be found. So, you can achieve the same with just one line.